dataset_conversion/nnunetV2_to_bids.py

```diff
@@ -0,0 +1,106 @@
+"""
+Converts nnUNetv2 dataset format to the BIDS-structured dataset. Full details about
+the format can be found here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md
+
+Théo Mathieu
+"""
+
+import argparse
+import numpy as np
+import shutil
+import pathlib
+from pathlib import Path
+import json
+import os
+import nibabel as nib
+
+def get_parser():
+    # parse command line arguments
+    parser = argparse.ArgumentParser(description='Convert BIDS-structured dataset to nnUNetV2 database format.')
+    parser.add_argument('--path-bids', required=True, help='Path to nnUNet dataset. Example: ~/data/dataset')
+    parser.add_argument('--path-out', required=True, help='Path to output directory. Example: ~/data/dataset-bids')
+    parser.add_argument('--suffix', required=True, help='Suffix of the label file Example: sub-003_T2w_SUFFIX.nii.gz')
+    parser.add_argument('--copy', '-cp', type=bool, default=False,
+                        help='Making symlink (False) or copying (True) the files in the Bids dataset. '
+                             'This option only affects the image file, the label file is copied regardless of the '
+                             ' option, default = False. Example for symlink: ø, Example for copy: --copy')
+    return parser
+
+
+
+def get_subject_info(file_name, contrast_dict):
+    """
+    Get different information about the current subject
+
+    Args:
+        file_name (str): Filename corresponding to the subject image
+        contrast_dict (dict): Dictionary, key channel_names from dataset.json
+
+    Returns:
+        sub_names (str): Name of the subject. Example: sub-milan002
+        ses (str): ses name
+        bids_nb (str): subject number in the BIDS dataset
+        info[2] (str): Contrast value in BIDS format. Example 0001
+        contrast (str): Image contrast (T2w, T1, ...)
+
+    """
+    name = file_name.split(".")[0]
+    info = name.split("_")
+    bids_nb = info[-2]
+    contrast = info[-1]
+    if len(info) == 4:
+        ses = info.pop(1)
+    else:
+        ses = None
+    sub_name = "_".join(info[:-2])
+    contrast = contrast.lstrip('0')
+    if contrast == '':
+        contrast = '0'
+    contrast_bids = contrast_dict[contrast]
+    return sub_name, ses, bids_nb, contrast, contrast_bids
+
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+    copy = args.copy
+    suffix =args.suffix
+    root = Path(os.path.abspath(os.path.expanduser(args.path_bids)))
+    path_out = Path(os.path.abspath(os.path.expanduser(args.path_out)))
+    with open(os.path.join(root, "dataset.json"), 'r') as json_file:
+        dataset_info = json.load(json_file)
```

Comment:
A check if the `dataset.json` file exists could be added, raising an error if the file does not exist.

Reply:
Done in https://github.com/ivadomed/utilities/pull/15/commits/5d0b9fcbb66554eb32982e42582eef28fa676e6b